src/geometry/adjoints.jl

```diff
@@ -87,14 +89,15 @@ function ChainRulesCore.rrule(::typeof(ImageTransformations.warp!), out, img::Ab
     function warp!_pb(Δy)
         ∇out = NoTangent()
         ∇img = @not_implemented("To be implemented.")
-        ∇tform = Tangent{DiffImages.Homography}(H = zeros(eltype(tform.H), 3, 3)) # TODO: Change this to Tangent{LinearMap} when generalized in future
+        flds = fieldnames(typeof(tform))
+        ∇tform = canonicalize(Tangent{typeof(tform)}())
         Δy = collect(Δy)
         for p in CartesianIndices(out)
             _, ∇τ = rrule(ImageTransformations._getindex, img, p.I)
             _, _, ∇τ = ∇τ(Δy[p])
             _, ∇ϕ = rrule_via_ad(Zygote.ZygoteRuleConfig(), tform, SVector(p.I))
             ∇h, _ = ∇ϕ(∇τ)
-            ∇tform += Tangent{DiffImages.Homography}(H = ∇h.H)
+            ∇tform += Tangent{typeof(tform)}(;NamedTuple{flds}(∇h)...)
```

Comment:
```suggestion
            ∇tform += Tangent{typeof(tform)}(NamedTuple{flds}(∇h)...)
```

Reply:
Can't do this xD. I wanted to pass the field names too, so have to pass it as kwargs :p
If I splat the namedtuple directly, the values get passed but not the keys.